element_calcium_imaging/imaging_report.py

```diff
@@ -0,0 +1,418 @@
+    return masks
+
+
+def alpha_combine_2images(image, masks):
+    alpha = 0.5
+    return np.uint8(image * alpha + masks * (1 - alpha))
+
+
+def make_maskid_image(img, roi_ids, mask_xpix, mask_ypix):
+    shape = img.shape[:2]
+    maskid_image = np.full(shape, -1)
+    for xpix, ypix, roi_id in zip(mask_xpix, mask_ypix, roi_ids):
+        maskid_image[ypix, xpix] = roi_id
+    return maskid_image
+
+
+def get_tracelayout(key, width=600, height=600):
+    text = f"Trace for Cell {key['mask']}" if isinstance(key, dict) else "Trace"
+
+    return dict(
+        margin=dict(l=0, r=0, b=0, t=65, pad=0),
+        width=width,
+        height=height,  # 700,
+        transition={"duration": 0},
+        title={
+            "text": text,
+            "xanchor": "center",
+            "yanchor": "top",
+            "y": 0.97,
+            "x": 0.5,
+        },
+        xaxis={
+            "title": "Time (sec)",
+            "visible": True,
+            "showticklabels": True,
+            "showgrid": True,
+        },
+        yaxis={
+            "title": "Fluorescence (a.u.)",
+            "visible": True,
+            "showticklabels": True,
+            "showgrid": True,
+            "anchor": "free",
+            "overlaying": "y",
+            "side": "left",
+            "position": 0,
+        },
+        yaxis2={
+            "title": "Calcium Event (a.u.)",
+            "visible": True,
+            "showticklabels": True,
+            "showgrid": True,
+            "anchor": "free",
+            "overlaying": "y",
+            "side": "right",
+            "position": 1,
+            "titlefont": dict(color="#d62728"),
+            "tickfont": dict(color="#d62728"),
+        },
+        shapes=[
+            go.layout.Shape(
+                type="rect",
+                xref="paper",
+                yref="paper",
+                x0=0,
+                y0=0,
+                x1=1.0,
+                y1=1.0,
+                line={"width": 1, "color": "black"},
+            )
+        ],
+        legend={
+            "traceorder": "normal",
+            "yanchor": "top",
+            "y": 0.99,
+            "xanchor": "right",
+            "x": 0.99,
+        },
+        plot_bgcolor="rgba(0,0,0,0.05)",
+        modebar_remove=[
+            "zoom",
+            "resetScale",
+            "pan",
+            "select",
+            "zoomIn",
+            "zoomOut",
+            "autoScale2d",
+        ],
+    )
+
+
+@schema
+class ScanLevelReport(dj.Computed):
+    definition = """
+    -> imaging.Segmentation
+    ---
+    average_image: longblob
+    """
+
+    def make(self, key):
+        image = (imaging.MotionCorrection.Summary & key).fetch1("average_image")
+
+        cell_mask_ids, mask_xpix, mask_ypix = (
+            imaging.Segmentation.Mask * imaging.MaskClassification.MaskType & key
+        ).fetch("mask", "mask_xpix", "mask_ypix")
+
+        background_image = single_to_3channel_image(image, 1, 99)
```

Comment:
2% clipping is usually too much. Maybe 0.1 and 99.9 would work better.

Reply:
Suite2p is clipping 1% (https://github.com/MouseLand/suite2p/blob/6cf53acbae19ef5d4f484c04bd6a3e763be5f01b/suite2p/detection/anatomical.py#L148-L194)

Reply:
that's way too much, especially on the bright side.

Reply:
What I typically do is apply a grayscale transformation, something like `image ** 0.5` and then clip at 0.999.

Reply:
Like this?
```
def single_to_3channel_image(image, low_q=0, high_q=99.9):
    image = (image-image.min())**0.5
    low_p, high_p = np.percentile(image, [low_q, high_q])
    image = np.uint8(255 * (image - low_p) / (high_p - low_p))
    return np.dstack([image] * 3)
```

Reply:
Yes, this should preserve more detail.

Reply:
OK. I made the change. This should resolve the conversation.